content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,228 +11,193 @@ weight: 10
+ 
+### Volume specification example
+ 
+First, specify a pod that will provide the volume. This example uses the `emptyDir` volume type, and includes mounting the volume. 
 
-#### AWS EBS configuration example
+A `Pod`'s specification defines what `volumes` are available to containers in that pod through the `.spec.volumes` field.
 
 ```yaml
 apiVersion: v1
 kind: Pod
 metadata:
-  name: test-ebs
+  name: sample-cache-pod
 spec:
-  containers:
-  - image: k8s.gcr.io/test-webserver
-    name: test-container
-    volumeMounts:
-    - mountPath: /test-ebs
-      name: test-volume
+  # other specification, including container definitions
   volumes:
-  - name: test-volume
-    # This AWS EBS volume must already exist.
-    awsElasticBlockStore:
-      volumeID: "<volume id>"
-      fsType: ext4
+    - name: cache-volume
+      emptyDir: {}
 ```
 
-If the EBS volume is partitioned, you can supply the optional field `partition: "<partition number>"` to specify which parition to mount on.
-
-#### AWS EBS CSI migration
-
-{{< feature-state for_k8s_version="v1.17" state="beta" >}}
-
-The `CSIMigration` feature for `awsElasticBlockStore`, when enabled, redirects
-all plugin operations from the existing in-tree plugin to the `ebs.csi.aws.com` Container
-Storage Interface (CSI) driver. In order to use this feature, the [AWS EBS CSI
-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationAWS`
-beta features must be enabled.
-
-#### AWS EBS CSI migration complete
-
-{{< feature-state for_k8s_version="v1.17" state="alpha" >}}
+Second, specify a mount point within each container via the
+`.spec.containers[*].volumeMounts` field. 
 
-To disable the `awsElasticBlockStore` storage plugin from being loaded by the controller manager
-and the kubelet, set the `CSIMigrationAWSComplete` flag to `true`. This feature requires the `ebs.csi.aws.com` Container Storage Interface (CSI) driver installed on all worker nodes.
+Volumes can not mount into other volumes, or have hard links to
+other volumes. Each container in the pod's configuration must independently specify where to mount each volume. 
 
-### azureDisk {#azuredisk}
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: sample-cache-pod
+spec:
+  volumes:
+    [...]
+  containers:
+    - name: test
+      image: busybox
+      volumeMounts:
+        - name: cache-volume
+          mountPath: /var/cache
 
-The `azureDisk` volume type mounts a Microsoft Azure [Data Disk](https://docs.microsoft.com/en-us/azure/aks/csi-storage-drivers) into a pod.
+```
 
-For more details, see the [`azureDisk` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_disk/README.md).
+### Core volume types {#volume-types-core}
```

Comment:
I think it may make more sense to divide the volume types by "kubernetes core" and "general purpose storage".

Kubernetes core are specialized volume types (mostly ephemeral) and are not used for general purpose persistent storage use cases. They include:
- configmap
- downwardapi
- emptydir
- gitrepo
- hostpath
- projected
- secret

The remaining types are all general purpose storage and most commonly used as PersistentVolumes

Reply:
Ok, I have been spinning my wheels on how to break this down. I think that "Kubernetes core" (i.e., Kubernetes specific), and "general purpose" are better categories. 

I had previously considered "externally hosted" and "native" but there were a number of pitfalls. However, I think "Kubernetes specific" will work well.

Reply:
I have implemented this change. I sincerely appreciate your insight on this particular issue, it has resolved a persistent (ha) problem I had with this PR.